snapcraft/extensions/env_injector.py

```diff
@@ -0,0 +1,135 @@
+    snap options into environment variables automatically.
+
+    - To set global environment variables for all applications **inside** the snap:
+    
+    .. code-block:: shell
+        sudo snap set <snap-name> env.<key>=<value>
+
+    - To set environment variables for a specific application **inside** the snap:
+
+    .. code-block:: shell
+        sudo snap set <snap-name> apps.<app-name>.env.<key>=<value>
+
+    - To set environment file inside the snap:
+
+    .. code-block:: shell
+        sudo snap set <snap-name> env-file=<path-to-env-file>
+
+    """
+
+    @staticmethod
+    @overrides
+    def get_supported_bases() -> Tuple[str, ...]:
+        return ("core24",)
+
+
+    @staticmethod
+    @overrides
+    def get_supported_confinement() -> Tuple[str, ...]:
+        return ("strict", "devmode", "classic")
+
+
+    @staticmethod
+    @overrides
+    def is_experimental(base: Optional[str]) -> bool:
+        return True
+    
+    @overrides
+    def get_root_snippet(self) -> Dict[str, Any]:
+        return {}
+
+    
+    @overrides
+    def get_app_snippet(self, *, app_name: str) -> Dict[str, Any]:
+        """Return the app snippet to apply."""
+        return {
+            "command-chain": [ "bin/command-chain/env-exporter"],
+            "environment": {
+                "env_alias": f"{app_name}",
+            }
+        }
+
+    @overrides
+    def get_part_snippet(self, *, plugin_name: str) -> Dict[str, Any]:
+        return {
+             "build-environment": [
+                 { "VARS_INJECT_EXTENSION": "true" }
+             ]
+        }
+
+    @overrides
+    def get_parts_snippet(self) -> Dict[str, Any]:
+        toolchain = self.get_toolchain()
+        return {
+            f"env-injector/env-injector": {
+                "source": f"{get_extensions_data_dir()}/env-injector",
+                "plugin": "nil",
+                "build-snaps": [
+                    f"rustup",
+                ],
+                "build-packages": [
+                    f"musl-tools",
+                    f"upx-ucl", # for binary compression
+                ],
+                "override-build": f"""
+
+      rustup default stable
+      rustup target add {toolchain}
+
+      cargo build --target {toolchain} --release
+      mkdir -p $SNAPCRAFT_PART_INSTALL/bin/command-chain
+
+      cp target/{toolchain}/release/env-exporter $SNAPCRAFT_PART_INSTALL/bin/command-chain
+
+      # shrink the binary
+      upx --best --lzma target/{toolchain}/release/env-exporter
+      cp target/{toolchain}/release/env-exporter $SNAPCRAFT_PART_INSTALL/bin/command-chain/env-exporter-upx
+
+                """,
+
+            }
+        }
+
+    def get_toolchain(self):
+        # Dictionary mapping architecture names
+        toolchain = {
+            'amd64': 'x86_64-unknown-linux-musl',
+            'arm64': 'aarch64-unknown-linux-musl',
+            'armhf': 'arm-unknown-linux-musleabihf',
```

Comment:
Snaps aren't supported on armv6. The `armhf` arch refers to armv7; change to `armv7-unknown-linux-musleabihf`.

https://doc.rust-lang.org/rustc/platform-support.html